programs/stake/src/stake_state.rs

```diff
@@ -687,6 +705,21 @@ impl Stake {
             Ok(())
         }
     }
+
+    fn reduce_overage(&mut self, balance: u64, rent_exempt_balance: u64) -> Option<(u64, u64)> {
+        // at this moment, balance >>> rent_excempt_balance (no overflow risk)
+        //   => no there could be possibility of non-rent-exempt stake accounts
+        // this will overwrite innocent deactivating and immeditealy withdrawn stake accounts as well is this ok?
+        let corrected_stake = balance.saturating_sub(rent_exempt_balance);
+        if self.delegation.stake > corrected_stake {
+            // this could result in creating 0 stake account..; is this safe?
```

Comment:
I *think* as long as 0 stake generates 0 rewards, it should be safe. How were you thinking we'd check this?

Reply:
> How were you thinking we'd check this?

I checked by use of :eyes:

Or, if you want extra safety, I think I can write tests for stake/inflation/split/merge/withdraw for 0 lamport accounts... A bit daunting work. ;)